src/types.rs

```diff
@@ -199,11 +200,15 @@ impl BoundedStorable for PrincipalStorable {
 
 #[derive(Clone, Debug, Eq, PartialEq, CandidType, Deserialize)]
 pub struct ProviderView {
+    #[serde(rename = "providerId")]
```

Comment:
Note that Candid doesn't support `#[serde(rename_all = ...)]`, so these individual attributes are necessary. Leaving out the attributes results in an encoding / decoding mismatch, so this is required for supporting both JSON and Candid (de)serialization from the same data structure.